resolve/resolve.go

```diff
@@ -505,7 +505,7 @@ func (r *resolver) stmt(stmt syntax.Stmt) {
 		}
 
 	case *syntax.IfStmt:
-		if !r.options.GlobalReassign && r.container().function == nil {
+		if !r.options.GlobalReassign && !r.options.TopLevelControl && r.container().function == nil {
```

Comment:
I think these three should just be `if !r.options.TopLevelControl ...`, since the GlobalReassign flag was split into two fine-grained options fields.

Reply:
Done.